pygmt/src/blockm.py

```diff
@@ -0,0 +1,215 @@
+    V="verbose",
+    a="aspatial",
+    f="coltypes",
+    r="registration",
+)
+@kwargs_to_strings(R="sequence")
+def blockmedian(table, outfile=None, **kwargs):
+    r"""
+    Block average (x,y,z) data tables by median estimation.
+
+    Reads arbitrarily located (x,y,z) triples [or optionally weighted
+    quadruples (x,y,z,w)] from a table and writes to the output a median
+    position and value for every non-empty block in a grid region defined by
+    the ``region`` and ``spacing`` parameters.
+
+    Full option list at :gmt-docs:`blockmedian.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    table : pandas.DataFrame or str
+        Either a pandas dataframe with (x, y, z) or (longitude, latitude,
+        elevation) values in the first three columns, or a file name to an
+        ASCII data table.
+
+    spacing : str
+        *xinc*\[\ *unit*\][**+e**\|\ **n**]
+        [/*yinc*\ [*unit*][**+e**\|\ **n**]].
+        *xinc* [and optionally *yinc*] is the grid spacing.
+
+    region : str or list
+        *xmin/xmax/ymin/ymax*\[\ **+r**\][**+u**\ *unit*].
+        Specify the region of interest.
+
+    outfile : str
+        Required if ``table`` is a file. The file name for the output ASCII
+        file.
+
+    {V}
+    {a}
+    {f}
+    {r}
+
+    Returns
+    -------
+    output : pandas.DataFrame or None
+        Return type depends on whether the ``outfile`` parameter is set:
+
+        - :class:`pandas.DataFrame` table with (x, y, z) columns if ``outfile``
+          is not set
+        - None if ``outfile`` is set (filtered output will be stored in file
+          set by ``outfile``)
+    """
+    return _blockm(block_method="blockmedian", table=table, outfile=outfile, **kwargs)
+
+
+def _blockm(block_method, table, outfile, **kwargs):
+    r"""
+    Block average (x,y,z) data tables by mean or median estimation.
+
+    Reads arbitrarily located (x,y,z) triples [or optionally weighted
+    quadruples (x,y,z,w)] from a table and writes to the output a mean or
+    median (depending on ``block_method``) position and value for every
+    non-empty block in a grid region defined by the ``region`` and ``spacing``
+    parameters.
+
+    Parameters
+    ----------
+    block_method : str
+        Name of the GMT module to call. Must be "blockmean" or "blockmedian".
+
+    table : pandas.DataFrame or str
+        Either a pandas dataframe with (x, y, z) or (longitude, latitude,
+        elevation) values in the first three columns, or a file name to an
+        ASCII data table.
+
+    outfile : str
+        Required if ``table`` is a file. The file name for the output ASCII
+        file.
+
+    spacing : str
+        *xinc*\[\ *unit*\][**+e**\|\ **n**]
+        [/*yinc*\ [*unit*][**+e**\|\ **n**]].
+        *xinc* [and optionally *yinc*] is the grid spacing.
+
+    region : str or list
+        *xmin/xmax/ymin/ymax*\[\ **+r**\][**+u**\ *unit*].
+        Specify the region of interest.
+
+    {V}
+    {a}
+    {f}
+    {r}
```

Comment:
Sorry, I didn't mean to explain all parameters. As this is an internal function, explaining the unique parameter `block_method` is enough. 
```suggestion
```

Reply:
I read the comment in email and did not notice the in-line suggestion. The removal was done in https://github.com/GenericMappingTools/pygmt/pull/1092/commits/55a97be3aad14bd177fa56eb972ea2df244cc2a1